contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -599,6 +599,12 @@ abstract contract PoSValidatorManager is
             });
             return (reward > 0);
         } else if (validator.status == ValidatorStatus.Completed) {
+            // To prevent churn tracker abuse, check that one full churn period has passed,
+            // so a delegator may not stake twice in the same churn period.
+            if (block.timestamp < delegator.startedAt + _getChurnPeriodSeconds()) {
+                revert MinStakeDurationNotPassed(uint64(block.timestamp));
+            }
```

Comment:
I think this check should actually be moved into `_completeEndDelegation` itself. Currently I think there is still a corner case where `_completeEndDelegation` could be called from `completeDelegatorRegistration` prior to a churn period elapsing if the validator completed prior the delegation registration being completed.

Reply:
Otherwise, we could also just add the similar check in that case, but feels more robust and less error prone to include it in a single place within `_completeEndDelegation` (possibly with the downside of doing a redundant check in certain cases).

Reply:
Ahh yep, you're right